packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -770,7 +770,7 @@ function shouldPreventDefault(target: Element) {
 }
 
 function shouldPreventDefaultKeyboard(target: Element) {
-  return !((target.tagName === 'INPUT' || target.tagName === 'BUTTON') && (target as HTMLButtonElement | HTMLInputElement).type === 'submit');
+  return !((target.tagName === 'INPUT' || target.tagName === 'BUTTON') && ((target as HTMLButtonElement | HTMLInputElement).type === 'submit' || (target as HTMLButtonElement | HTMLInputElement).type === 'reset'));
```

Comment:
What do we think about this change? Do we want to add `onReset` to Form as well to mirror `onSubmit`? Or do we prefer that the user do the following approach for their reset buttons: https://github.com/adobe/react-spectrum/blob/dfe72ee076f7ca27728d0b6b4c1ef86e11191ece/packages/%40react-spectrum/form/stories/Form.stories.tsx#L575

Reply:
Ok, so there are some issues with form reset in React, have you taken a look at my old PR that addressed submit? it used to handle reset as well, and if I recall correctly, the radio group and a few other controlled components just couldn't handle reset

Reply:
I found this following thread: https://github.com/adobe/react-spectrum/pull/904#discussion_r467188676. Did some digging online and I see mention of controlled components having problems with reset and for the most part this is handled via a onReset callback. I can punt on this PR until we add onReset or I can implement it here, but IMO this at least fixes the uncontrolled case and doesn't break any additional functionality

Reply:
yeah, it's fine to add it, it looks like we also need to add a Type to Forms for onReset?

I think the hesitation for adding it was that even people who are using our components uncontrolled would run into issues. For instance, NumberField and RadioGroup are technically controlled in our implementation. People can choose to use our components in a controlled or uncontrolled way, but under the hood, WE are always using them in a controlled way. And during a reset, no event is fired, so our components don't know to update.

Reply:
makes sense, I'll see about putting this all into an issue. Def can document the above in the Forms docs